pkg/alerts/prometheus.go

```diff
@@ -7,22 +7,40 @@ import (
 	"sort"
 	"time"
 
-	"github.com/prometheus/client_golang/api"
+	papi "github.com/prometheus/client_golang/api"
 	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
 	"github.com/prometheus/common/model"
 )
 
-// PrometheusActiveAlerts returns a list of names of active (e.g. pending or firing) alerts, filtered
-// by the supplied regexp.
-func PrometheusActiveAlerts(prometheusURL string, filter *regexp.Regexp) ([]string, error) {
-	client, err := api.NewClient(api.Config{Address: prometheusURL})
+// PromClient is a wrapper around the Prometheus Client interface and implements the api
+// This way, the PromClient can be instantiated with the configuration the Client needs, and
+// the ability to use the methods the api has, like Query and so on.
+type PromClient struct {
+	papi papi.Client
+	api  v1.API
+}
+
+// New creates a new client to the Prometheus API.
+// It returns an error on any problem.
+func (p PromClient) New(conf papi.Config) (*PromClient, error) {
```

Comment:
I think we want an exportable `New()` func which returns a new instance, rather than a method against the type struct that does so. So this instead:

```go
func New(conf papi.Config) (*PromClient, error) {
        promClient, err := papi.NewClient(conf)
 	if err != nil {
 		return nil, err
 	}
 	client := PromClient{papi: promClient, api: v1.NewAPI(promClient)}
 	return &client, nil
}
```

Reply:
I agree on this one. fixed it.